src/server/pfs/server/driver.go

```diff
@@ -1124,6 +1097,11 @@ func (d *driver) makeCommit(
 		if err != nil {
 			return nil, err
 		}
+		// there should only be one representative of each branch in the commit provenance
+		if _, ok := provenantBranches[key(prov.Branch.Repo.Name, prov.Branch.Name)]; ok {
+			return nil, fmt.Errorf("the commit provenance contains multiple commits provenant on the same branch")
```

Comment:
The commits aren't really provenant on the same branch so much as they're from the same branch right?

Reply:
oh right

Reply:
reworded it